File: solutions/zig/sublist/1/sublist.zig
// For Shree DR.MDD

pub const Relation = enum {
    equal,
    sublist,
    superlist,
    unequal,
};

fn checkSublist(smaller: []const i32, bigger: []const i32) bool {
    if (smaller.len == 0) return true;
    if (smaller.len > bigger.len) return false;

    outer_loop: for (0..(bigger.len - smaller.len + 1)) |start| {
        for (smaller, 0..) |val, idx| {
            if (val != bigger[start + idx]) continue :outer_loop;
        }
        return true;
    }

    return false;
}

pub fn compare(list_a: []const i32, list_b: []const i32) Relation {
    if (list_a.len == list_b.len) {
        for (list_a, list_b) |x, y| {
            if (x != y) return .unequal;
        }
        return .equal;
    }

    if (list_a.len < list_b.len) {
        return if (checkSublist(list_a, list_b)) .sublist else .unequal;
    } else {
        return if (checkSublist(list_b, list_a)) .superlist else .unequal;
    }
}

File: solutions/lua/sublist/1/sublist.lua
local function find(l1, l2)
  if #l1 == 0 then return true end
  for i = 1, #l2 - #l1 + 1 do
    if l2[i] == l1[1] then
      local found = true
      for j = 2, #l1 do
        if l1[j] ~= l2[i+j-1] then
          found = false
          break
        end
      end
      if found then return true end
    end
  end
  return false
end
return function(l1, l2)
  if find(l1, l2) then
    return #l1 == #l2 and "equal" or "sublist"
  end
  return find(l2, l1) and "superlist" or "unequal"
end